gopherage/pkg/cov/junit/calculation/coveragelist.go

```diff
@@ -43,8 +43,6 @@ func (covList *CoverageList) Ratio() float32 {
 
 // summarize summarizes all items in the Group and stores the result
 func (covList *CoverageList) summarize() {
-	covList.NumCoveredStmts = 0
```

Comment:
if we don't zero this, repeated calls are going to keep summing higher and higher?

Reply:
Each object is new each time `s := newCoverageList("Filtered Summary: " + prefix)` so we will never iterate over the same one twice. That being said. If we did, I would think we wouldn't want to re-zero so that a subdir contains the results for all coverage results with that matching prefix.

Reply:
https://github.com/kubernetes/test-infra/blob/2035c42220f2ab250958f28c78255d24f895e949/gopherage/pkg/cov/junit/xmlwriter.go#L85-L86

Reply:
> Each object is new each time s := newCoverageList("Filtered Summary: " + prefix) so we will never iterate over the same one twice. 

In the current code. But as an API it doesn't make sense to repeatedly add to the same fields if called twice.

> // summarize summarizes all items in the Group and stores the result

I would not expect two calls to summarize() to produce a 2x count when accessed.
One way or another this clearly needs refactoring.

Reply:
I added them back in. based on the doc for that method I think reseting makes sense. Just because it currently isn't recalled. Doesn't mean it wont ever be!